### PR TITLE
Refactor manifest

### PR DESCRIPTION
**Pull request by ruben-arts**
Refactoring the `Project` into `Manifest`

This is done to split the responsibilities.

### CODE REVIEW

File: src/project/mod.rs [hunk 1]
```diff
@@ -870,7 +389,7 @@ impl Project {
 
         // Gather the main activation scripts if there are no target scripts defined.
         if all_scripts.is_empty() {
-            if let Some(activation) = &self.manifest.activation {
+            if let Some(activation) = &self.manifest.manifest.activation {
```
Comment (wolfv) [inline src/project/mod.rs:392]: the manifest.manifest duplication looks a bit funky. can we rename one of them?

File: tests/common/mod.rs [hunk 2]
```diff
@@ -170,7 +170,7 @@ impl PixiControl {
 
     /// Loads the project manifest and returns it.
     pub fn project(&self) -> miette::Result<Project> {
-        Project::load(&self.manifest_path())
+        Project::load_or_else_discover(Some(&self.manifest_path()))
```
Comment (wolfv) [inline tests/common/mod.rs:173]: nice rename!

File: src/project/manifest.rs [hunk 3]
```diff
@@ -14,8 +17,554 @@ use serde_with::{serde_as, DeserializeAs, DisplayFromStr, PickFirst};
...
+    pub path: PathBuf,
+
+    /// The raw contents of the manifest file
+    pub contents: String,
+
+    /// Editable toml document
+    pub document: toml_edit::Document,
+
+    /// The parsed manifest
+    pub manifest: ProjectManifest,
```
Comment (wolfv) [inline src/project/manifest.rs:41]: should this be called `parsed` or something like this? Might make the distinction more obvious.

**Review by wolfv — APPROVED** [review]
Nice stuff. Left some comments

File: src/project/manifest.rs (hunk 3 above)
Comment (ruben-arts) [reply src/project/manifest.rs:41]: Good one, let me discuss with gary

**Review by baszalmstra — APPROVED** [review]

File: src/project/manifest.rs (hunk 3 above)
Comment (ruben-arts) [reply src/project/manifest.rs]: Renamed it to that!

File: src/project/mod.rs (hunk 1 above)
Comment (ruben-arts) [reply src/project/mod.rs]: it's now `manifest.parsed`

File: src/project/manifest.rs [hunk 4]
```diff
@@ -14,8 +17,554 @@ use serde_with::{serde_as, DeserializeAs, DisplayFromStr, PickFirst};
...
 use std::ops::Range;
 use std::path::{Path, PathBuf};
+use toml_edit::{Array, Document, Item, Table, TomlError, Value};
 use url::Url;
 
+/// Handles the project's manifest file.
+/// This struct is responsible for reading, parsing, editing, and saving the manifest.
+/// It encapsulates all logic related to the manifest's TOML format and structure.
+/// The manifest data is represented as a [`ProjectManifest`] struct for easy manipulation.
+/// Owned by the [`Project`] struct, which governs its usage.
```
Comment (tdejager) [inline src/project/manifest.rs:27]: Guess you need the entire path :)

File: src/project/manifest.rs [hunk 5]
```diff
@@ -14,8 +17,554 @@ use serde_with::{serde_as, DeserializeAs, DisplayFromStr, PickFirst};
...
+    /// Editable toml document
+    pub document: toml_edit::Document,
+
+    /// The parsed manifest
+    pub parsed: ProjectManifest,
+}
+
+impl Manifest {
+    /// Create a new manifest from a path
+    pub fn from_path(path: PathBuf) -> miette::Result<Self> {
```
Comment (tdejager) [inline src/project/manifest.rs:46]: I guess I would just pass it a &Path and make it owned in the function :)

File: src/project/manifest.rs [hunk 6]
```diff
@@ -14,8 +17,554 @@ use serde_with::{serde_as, DeserializeAs, DisplayFromStr, PickFirst};
...
+            parsed: manifest,
+        })
+    }
+
+    /// Save the manifest to the file and update the contents
+    pub fn save(&mut self) -> miette::Result<()> {
+        self.contents = self.document.to_string();
+        std::fs::write(&self.path, self.contents.clone()).into_diagnostic()?;
+        Ok(())
+    }
```
Comment (tdejager) [inline src/project/manifest.rs:103]: Can't see from github PR but does `&self` need to be mutable?

File: tests/common/mod.rs (hunk 2 above)
Comment (tdejager) [reply tests/common/mod.rs:174]: Indeed!

**Review by tdejager — CHANGES_REQUESTED** [review]

File: src/project/manifest.rs (hunk 6 above)
Comment (ruben-arts) [reply src/project/manifest.rs:103]: I'm overwriting the `self.contents` that is what i needed the mutability for.

File: src/project/manifest.rs (hunk 5 above)
Comment (baszalmstra) [reply src/project/manifest.rs:46]: But then you might introduce an additional copy right? If you have a `PathBuf` already, you take its reference and then clone it in the function. I think if the function is going to take ownership it makes sense the pass the `PathBuf` straight away. Or use `impl Into<PathBuf>`.

Comment (ruben-arts) [reply src/project/manifest.rs]: Changed it!

Comment (baszalmstra) [reply src/project/manifest.rs]: Scratch my comment, the function doesn't take ownership of the `path`.

Comment (ruben-arts) [reply src/project/manifest.rs]: Oh only now see @baszalmstra comment.

Comment (ruben-arts) [reply src/project/manifest.rs]: Moved it to: 
```rust
pub fn from_path(path: impl AsRef<Path>) -> miette::Result<Self> 
```
As this should avoid unneeded copying and I added a test that shows it is working + also works with multiple types.

**Review by tdejager — APPROVED** [review]

**Pull request merged**